.clj-kondo/config.edn

```diff
@@ -83,7 +83,9 @@
     schema.core/either                                                           {:namespaces ["metabase.*"]}}}
 
   :discouraged-var
-  {toucan.db/query           {:message "Use mdb.query/query instead of toucan.db/query"}
+  {clojure.string/lower-case {:message "Use metabase.util/lower-case-en instead of clojure.string/lower-case"}
+   clojure.string/upper-case {:message "Use metabase.util/upper-case-en instead of clojure.string/upper-case"}
```

Comment:
I'm not sure we should use English everywhere. If we want to make people pick the right function intentionally, we might have to introduce another pair which is correct when we want to use the users or the systems locale.

Reply:
I think the idea is that in bespoke cases we'll have a `:clj-kondo/ignore`, and that it signifies the dev thought about it and is using `str/lower-case` by design and not out of ignorance.

Reply:
> If we want to make people pick the right function intentionally, we might have to introduce another pair which is correct when we want to use the users or the systems locale.

👍 I think adding functions that use `site-locale` or `user-locale` make sense. We probably shouldn't be using `str/lower-case` or `str/upper-case` at all, since they use the _system locale_, which is not necessarily the same as the `site-locale` Setting. For example maybe I'm running Metabase on AWS and the system locale is English, but I want the default site locale for Metabase to be something else, so I set `site-locale` Setting. `site-locale` Setting != JVM system locale

Reply:
Until we add the ability to customize locale, should we leave this as-is. In the future the warning could be: "Use localized string functions (e.g. metabase.util/lower-case-en) instead of clojure.string/lower-case"

Reply:
At least at those places where we are formatting for the user, I wouldn't switch to `en`. I would prefer keeping backward compatibility (and using the system locale) or switching to the user locale (which might be too much for this PR).

Reply:
> Until we add the ability to customize locale

FWIW you could do it already with Java interop with something like `(.toLowerCase (str whatever) (i18n/site-locale))`, but it would be pretty easy to put that into a function